### ISSUE TITLE
BloomEffect with selective objects

### ISSUE THREAD

**Issue by mrgoonie** [vanruesc/postprocessing#142]
Hi @vanruesc, great library! 

I'm trying to achieve **Bloom Effect** on some selective objects on the scene only.

Tried to combine [this example](https://threejs.org/examples/#webgl_postprocessing_unreal_bloom_selective) with your library but haven't succeeded so far. 

Can we have something like `setSelection` &`selectObject` function with **Bloom Effect** just like the `OutlineEffect`? 😄 

Thanks,

**Comment by vanruesc** [comment]
Hi!

> Can we have something like `setSelection` & `selectObject` function with Bloom Effect just like the `OutlineEffect`?

Yes, that feature should be easy to add.

I'm currently working on upgrading `postprocessing` to the latest version of `three`. You can expect selective bloom in the release after the next one.

**Comment by mrgoonie** [comment]
Great to hear that! Thanks, mate.

**Comment by vanruesc** [comment]
FYI: I haven't forgotton about this feature request. Just been a bit busy lately.

> that feature should be easy to add.

The implementation will require more time than anticipated. I'm planning on moving the selective render feature to the `RenderPass` to avoid code duplication in the `OutlinePass`.  Not sure when I'll get a chance to work on this, but it shouldn't be too long.

**Comment by vanruesc** [comment]
Added `SelectiveBloomEffect` in `postprocessing@6.7.0`.

The selection feature could have been added to the existing `BloomEffect`, but I decided to create a new effect instead which extends `BloomEffect`. The selection logic requires additional rendering which has an impact on performance, so I kept it out of the standard bloom solution.

The disadvantage in performance comes from the fact that all objects in the scene need to be rendered again: non-selected objects are rendered solid black to properly occlude selected objects and the scene background.

The `OutlineEffect` has the same fundamental problem. A while ago, I managed to remove the additional rendering step from the `GodRaysEffect` by relying on depth, but that was only possible because the light source could be treated as a special object and the same method can't be applied to either the `OutlineEffect` or the `SelectiveBloomEffect`.

If you have any questions or ideas, lmk.
